sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -486,6 +486,9 @@ Scheduler::GraphBuilder::addCopyBack(Requirement *Req,
 
   std::vector<Command *> ToCleanUp;
   for (Command *Dep : Deps) {
+    if (Dep->MEnqueueStatus == EnqueueResultT::SyclEnqueueFailed)
```

Comment:
this is too common change and could happen in the middle of program. I think it is not correct

Reply:
It is definitely correct. I'm about to separate this out into its own PR, but what is happening is that we are enqueuing a set of dependencies/requirements and a command, during which an exception is thrown (and caught and rethrown).  This sets the result to EnqueuFailed, but only for that particular item, not for things that depend on it (or vice versa). The Command pointers are all loose pointers, not smart pointers. So at the time of the exception some of that set is already in the DAG and then abandoned, and they leak.   There is no other way to get the EnqueueFailed status. 

So the fix is to be more diligent about checking for EnqueuFailed and avoid the leak that way.  And this strategy applies to the buffer destructors as well.